ggshield/cmd/hmsl/fingerprint.py

```diff
@@ -0,0 +1,174 @@
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Any, Callable, Dict, Iterator, List, Optional, Set, TextIO, cast
+
+import click
+
+from ggshield.cmd.common_options import add_common_options
+from ggshield.core.filter import censor_string
+from ggshield.core.text_utils import display_info
+from ggshield.hmsl.client import PREFIX_LENGTH
+from ggshield.hmsl.crypto import hash_string
+
+
+# Types and constants
+
+
+class SourceType(Enum):
+    FILE = auto()
+
+
+@dataclass
+class PreparedSecrets:
+    payload: Set[str]
+    mapping: Dict[str, str]
+
+
+@dataclass
+class SecretWithKey:
+    key: Optional[str]
+    value: str
+
+
+# Methods to compute names for secrets
+# They are useful to help the user identify the secret that may leaked
+# with a more "human-readable" string than a hash.
+# Takes the secret and optional key as input and returns a string.
+NamingStrategy = Callable[[SecretWithKey], str]
+
+NAMING_STRATEGIES: Dict[str, NamingStrategy] = {
+    "censored": lambda secret: censor_string(secret.value),
+    "cleartext": lambda secret: secret.value,
+    "none": lambda _: "",
+    "key": lambda secret: secret.key or censor_string(secret.value),
+}
+
+
+# Command
+
+input_arg = click.argument(
+    "path",
+    type=click.Path(
+        exists=True,
+        dir_okay=False,
+        readable=True,
+        allow_dash=True,
+    ),
```

Comment:
You can use `path_type=Path` here this way the argument is a `Path` instance instead of an `str`. It's more expressive, and you get handy methods like `.open()` or `.read_text()`.

Reply:
does it still works with stdin? (`allow_dash=True`)